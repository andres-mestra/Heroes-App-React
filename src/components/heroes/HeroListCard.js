import * as React from 'react'
import PropTypes from 'prop-types'

export const HeroListCard = ({ 
  id,
  superhero,
  publisher,
  alter_ego,
  first_appearance,
  characters, 
}) => {
  return (
    <div>
      { superhero }
    </div>
  )
}

HeroListCard.propTypes = {
  id: PropTypes.oneOfType([ PropTypes.string, PropTypes.number ]).isRequired,
  superhero: PropTypes.string,
  publisher: PropTypes.string,
  alter_ego: PropTypes.string,
  first_appearance: PropTypes.string,
  characters: PropTypes.string, 
}