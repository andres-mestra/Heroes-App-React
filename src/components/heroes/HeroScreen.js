import * as React from 'react'
import { Redirect, useParams } from 'react-router-dom'
import { getHeroById } from '../../selectors/getHeroById';

export const HeroScreen = ({ history }) => {
  
  //optiene los argumentos de la url
  const { heroeId } = useParams();
  //Solo se vuelve a ejecutar si el heroeId cambia 
  const hero = React.useMemo(() => getHeroById( heroeId ),
  [heroeId])

  
  //Redirecc√≠ona si el hero no existe
  if( !hero ){
    return <Redirect to="/" />
  }

  const { 
    superhero,
    publisher,
    alter_ego,
    first_appearance,
    characters, 
  } = hero;

  const handleReturn = () => {
    //goBack: retorna a la pagina anterior
    (history.lengtn <= 2) ? history.push('/') : history.goBack();
  }

  return (
    <section className="row mt-5" >
      <div className="col-4">
        <img 
          src={`../assets/heroes/${heroeId}.jpg`}
          alt={superhero}
          className="img-thumbnail"
        />
      </div>
      <div className="col-8">
        <h1>{ superhero }</h1>
        <ul className="list-group list-group-flush">
          <li className="list-group-item"><b>Alter ego:</b> {alter_ego} </li>
          <li className="list-group-item"><b>Publisher:</b> {publisher} </li>
          <li className="list-group-item"><b>First appearance:</b> {first_appearance} </li>
        </ul>
        <h5> Characters </h5>
        <p>{ characters }</p>

        <button 
          className="btn btn-outline-info" 
          onClick={handleReturn}
        >
          Return
        </button>
      </div>
    </section>
  )
}
