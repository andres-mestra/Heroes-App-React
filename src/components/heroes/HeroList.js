import * as React from 'react'
import { getHeroesByPublisher } from '../../selectors/getHeroesByPublisher'
import { HeroListCard } from './HeroListCard';
import PropTypes from 'prop-types'


export const HeroList = ({ publisher }) => {
  //Solo se vuelve a ejecutar si el publisher cambia 
  const heroes = React.useMemo(() => getHeroesByPublisher( publisher ),
  [publisher])
  

  return (
    <section className="row justify-content-center animate__animated animate__fadeInRight" >
      {
        heroes.map( hero => (
          <HeroListCard 
            key={hero.id}
            {...hero}
          />
        ))
      }
    </section>
  )
}

HeroList.propTypes = {
  publisher: PropTypes.string.isRequired,
}