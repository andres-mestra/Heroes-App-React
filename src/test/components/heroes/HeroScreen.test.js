import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { mount } from 'enzyme'
import { HeroScreen } from '../../../components/heroes/HeroScreen'


describe('Pruebas en <HeroScreen />', () => {
  
  const historyMock = {
    lengtn: 10,
    push: jest.fn(),
    goBack: jest.fn(),
  }

  const wrapper = mount(
    <MemoryRouter initialEntries={['/hero']}>
      <HeroScreen history={historyMock} />
    </MemoryRouter>
  );

  test('debe mostrar el componente redirect si no hay argumento en la url', () => {
    
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('Redirect').exists() ).toBe(true);
  })
  
  
})
