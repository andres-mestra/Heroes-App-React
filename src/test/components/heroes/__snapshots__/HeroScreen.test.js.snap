// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroScreen /> debe mostrar el componente redirect si no hay argumento en la url 1`] = `""`;
